### PR TITLE
fix: Fix LazyServiceFactory to handle exception in constructor

### PR DESCRIPTION
**Pull request by idsulik**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | yes
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description
This PR fixes a bug where exceptions thrown during lazy service initialization would leave the proxy in an invalid state. The fix ensures that the original initializer is restored when an exception occurs, allowing subsequent initialization attempts to work correctly.
Added tests to verify the behavior.

p.s. thanks to @vdmorozov, he found the issue

### CODE REVIEW

**Comment by idsulik** [comment]
if it's possible, please create a patch release for the 3 version with this fix

File: src/Proxy/LazyServiceFactory.php [hunk 1]
```diff
@@ -42,8 +43,14 @@ public function __invoke(
...
         if (isset($this->servicesMap[$name])) {
             $initializer = static function (&$wrappedInstance, LazyLoadingInterface $proxy) use ($callback): bool {
+                $initializer = $proxy->getProxyInitializer();
                 $proxy->setProxyInitializer(null);
-                $wrappedInstance = $callback();
+                try {
+                    $wrappedInstance = $callback();
+                } catch (Throwable $e) {
+                    $proxy->setProxyInitializer($initializer);
+                    throw $e;
```
Comment (vdmorozov) [inline src/Proxy/LazyServiceFactory.php:52]: What if we just swap these two rows: `$proxy->setProxyInitializer(null)` and `$wrappedInstance = $callback()`?

It'd look like this:
```php
                $wrappedInstance = $callback();
                $proxy->setProxyInitializer(null);
```

**Comment by idsulik** [comment]
@vdmorozov will create a separate PR

**Comment by vdmorozov** [comment]
Here: https://github.com/laminas/laminas-servicemanager/pull/250

**Pull request closed without merging**